ingest_api/src/main.py

```diff
@@ -0,0 +1,219 @@
+)
+async def list_ingestions(
+    list_request: schemas.ListIngestionRequest = Depends(),
+    db: services.Database = Depends(dependencies.get_db),
+):
+    """
+    Lists the STAC items from ingestion.
+    """
+    return db.fetch_many(
+        status=list_request.status, next=list_request.next, limit=list_request.limit
+    )
+
+
+@app.post(
+    "/ingestions",
+    response_model=schemas.Ingestion,
+    tags=["Ingestion"],
+    status_code=201,
+)
+async def enqueue_ingestion(
+    item: schemas.AccessibleItem,
+    username: str = Depends(auth.get_username),
+    db: services.Database = Depends(dependencies.get_db),
+) -> schemas.Ingestion:
+    """
+    Queues a STAC item for ingestion.
+    """
+    return schemas.Ingestion(
+        id=item.id,
+        created_by=username,
+        item=item,
+        status=schemas.Status.queued,
+    ).enqueue(db)
+
+
+@app.get(
+    "/ingestions/{ingestion_id}",
+    response_model=schemas.Ingestion,
+    tags=["Ingestion"],
+)
+def get_ingestion(
+    ingestion: schemas.Ingestion = Depends(dependencies.fetch_ingestion),
+) -> schemas.Ingestion:
+    """
+    Gets the status of an ingestion.
+    """
+    return ingestion
+
+
+@app.patch(
+    "/ingestions/{ingestion_id}",
+    response_model=schemas.Ingestion,
+    tags=["Ingestion"],
+)
+def update_ingestion(
+    update: schemas.UpdateIngestionRequest,
+    ingestion: schemas.Ingestion = Depends(dependencies.fetch_ingestion),
+    db: services.Database = Depends(dependencies.get_db),
+):
+    """
+    Updates the STAC item with the provided item.
+    """
+    updated_item = ingestion.copy(update=update.dict(exclude_unset=True))
+    return updated_item.save(db)
+
+
+@app.delete(
+    "/ingestions/{ingestion_id}",
+    response_model=schemas.Ingestion,
+    tags=["Ingestion"],
+)
+def cancel_ingestion(
+    ingestion: schemas.Ingestion = Depends(dependencies.fetch_ingestion),
+    db: services.Database = Depends(dependencies.get_db),
+) -> schemas.Ingestion:
+    """
+    Cancels an ingestion in queued state."""
+    if ingestion.status != schemas.Status.queued:
+        raise HTTPException(
+            status_code=400,
+            detail=(
+                "Unable to delete ingestion if status is not "
+                f"{schemas.Status.queued}"
+            ),
+        )
+    return ingestion.cancel(db)
+
+
+@app.post(
+    "/collections",
+    tags=["Collection"],
+    status_code=201,
+    dependencies=[Depends(auth.get_username)],
+)
+def publish_collection(collection: schemas.DashboardCollection):
+    """
+    Publish a collection to the STAC database.
+    """
+    # pgstac create collection
+    try:
+        collection_publisher.ingest(collection)
+        return {f"Successfully published: {collection.id}"}
+    except Exception as e:
+        raise HTTPException(
+            status_code=400,
+            detail=(f"Unable to publish collection: {e}"),
+        )
```

Comment:
I wonder if we should handle collection ingestion via the whole dynamodb -> lambda -> pgstac workflow too?
Although publishing a collection shouldn't come in bulk like items can, and maybe it should happen synchronously 🤔 
Something to think about